### ISSUE TITLE
@LocatorApplication does not work with @EnableSecurity

### ISSUE THREAD

**Issue by janmaslik** [spring-projects/spring-boot-data-geode#118]
```
@UseLocators
@SpringBootApplication
@LocatorApplication(name = "Locator")
@EnableSecurity
public class GeodeLocatorApplication  
{
	public static void main(String[] args) 
        {
		new SpringApplicationBuilder(GeodeLocatorApplication.class)
			.web(WebApplicationType.NONE)
			.build()
			.run(args);

		System.err.println("Press <enter> to exit!");

		new Scanner(System.in).nextLine();
	}

}
```
using 
```
<groupId>org.springframework.geode</groupId>
<artifactId>spring-geode</artifactId>
<version>1.6.9</version>
```

Error: 
A component required a bean named 'gemfireCache' that could not be found.

**Comment by janmaslik** [comment]
i found workarround
```
@Bean
	public LocatorConfigurer securityConfigurer() 
	{
		return (beanName, locatorFactoryBean) -> {
			Properties currentProperties = locatorFactoryBean.getGemFireProperties();
			currentProperties.setProperty("security-manager", "com.shirtplatform.cloud.locator.AppSecurityManager");
			locatorFactoryBean.setGemFireProperties(currentProperties);
		};
	}
```

but still problem if there is secured another Locator :
 org.apache.geode.security.AuthenticationRequiredException: Failed to find credentials from [.....]

**Comment by jxblum** [comment]
@janmaslik - I was able to reproduce your issue.

```
...
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named 'gemfireCache' available
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeanDefinition(DefaultListableBeanFactory.java:862)
	at org.springframework.data.gemfire.config.annotation.ApacheShiroSecurityConfiguration.lambda$shiroGemFireBeanFactoryPostProcessor$1(ApacheShiroSecurityConfiguration.java:126)
...
..
.
```

FYI, the bits for [bootstrapping](https://docs.spring.io/spring-data/geode/docs/current/api/org/springframework/data/gemfire/config/annotation/LocatorApplication.html) an Apache Geode (or alternatively, VMW Tanzu GemFire) _Locator_ and [enabling](https://docs.spring.io/spring-data/geode/docs/current/api/org/springframework/data/gemfire/config/annotation/EnableSecurity.html) Apache Geode Security (Auth) technically resides in _Spring Data for Apache Geode_ (SDG), upon which _Spring Boot for Apache Geode_ (SBDG), this project, is based. SBDG does offer a few extensions, but the primary logic is in SDG.

In any case, I will provide a bit of feedback here, but will be addressing this issue in the SDG project ([here](https://github.com/spring-projects/spring-data-geode)) across versions `2.6` (SBDG `1.6` is [based on](https://github.com/spring-projects/spring-boot-data-geode/wiki/Spring-Boot-for-Apache-Geode-and-VMware-Tanzu-GemFire-Version-Compatibility-Matrix#supported-versions) SDG `2.6`, FYI), `2.7` and (currently) `3.0`. After the next SDG patch release is out, SBDG will pick up the changes in SDG in its respective versions.

Based on the configuration you shared above, it would appear that you are only trying to configure and use a custom Apache Geode `SecurityManager` ([_Javadoc_](https://geode.apache.org/releases/latest/javadoc/org/apache/geode/security/SecurityManager.html)) implementation.

The error you are encountering is caused by the presence of **Apache Shiro** on your _Spring Boot_ / _Spring Data Geode_ application classpath. The `NoSuchBeanDefinitionException` is thrown from [here](https://github.com/spring-projects/spring-data-geode/blob/2.6.5/spring-data-geode/src/main/java/org/springframework/data/gemfire/config/annotation/EnableSecurity.java#L51), and specifically [here](https://github.com/spring-projects/spring-data-geode/blob/2.6.5/spring-data-geode/src/main/java/org/springframework/data/gemfire/config/annotation/ApacheShiroSecurityConfiguration.java#L126).

If you are not using Apache Shiro Security [integration](https://docs.spring.io/spring-data/geode/docs/current/reference/html/#bootstrap-annotation-config-security) with Apache Geode, then you can do 1 of the following:

1. Remove Apache Shiro from the Spring Boot/Data application classpath.
2. Or, alternatively set the following property in _Spring Boot_ `application.properties` to `false`:

```properties
# Spring Boot application.properties.

spring.data.gemfire.security.shiro.enabled=false
```

You can only use 1 security provider (e.g. either Apache Shiro, or an implementation of Apache Goede's `SecurityManager` interface) anyway.

It appears you are using a custom `SecurityManager` implementation, hence: `com.shirtplatform.cloud.locator.AppSecurityManager`.

Additionally, and though possible, you should never have to set the `SecurityManager` implementation using the `LocatorConfigurer`.

You can simply declare:

```java
@UseLocators
@SpringBootApplication
@LocatorApplication(name = "Locator")
@EnableSecurity(securityManagerClassName = "com.shirtplatform.cloud.locator.AppSecurityManager")
public class GeodeLocatorApplication  { 
  // ...
}
```

Or, better yet:

```java
@UseLocators
@SpringBootApplication
@LocatorApplication(name = "Locator")
@EnableSecurity(securityManagerClass = com.shirtplatform.cloud.locator.AppSecurityManager.class)
public class GeodeLocatorApplication  { 
  // ...
}
```

See [here](https://docs.spring.io/spring-data/geode/docs/current/api/org/springframework/data/gemfire/config/annotation/EnableSecurity.html#securityManagerClass--).

This should allow you to proceed foward.

The connection to other secure _Locators_ seems to be a configuration problem. But, it is hard to say without know more about your configuration.  All "secure" Locators need to be enabled with security and share the credentials required to connect. However, this is more of a function of your `SecurityManager` than anything else.

**Comment by jxblum** [comment]
Sorry for the delay. A few updates:

1) First, I had to proceed forward in releasing SBDG `1.6.11` and `1.7.3` based on the current _Spring Boot_ `2.6.x` and `2.7.x` release schedules, ahead of the required changes (fix) for this issue.

2) Second, I have finally [completed the required changes](https://github.com/spring-projects/spring-data-geode/issues/621) in _Spring Data for Apache Geode_ (SDG) `2.6.x` and `2.7.x`.  The fixes will be included in the upcoming releases of SDG (i.e. SDG `2.6.7` and SDG `2.7.3`) [scheduled](https://calendar.spring.io/) mid next month, tentatively on Friday, September 16th, 2022. 

3) Subsequently, this means when Spring Boot 2.6.12 and 2.7.4 are released following Spring Data, ~Thursday, September 22nd, 2022, Spring Boot for Apache Geode `1.6.12` and `1.7.4` will follow suit and release as well, which will include the new versions of SDG (respectively; i.e. `2.6.7` and `2.7.3`) which contain the required changes and fixes for this issue.

This turned out to be a rather involved set of changes.  However, after careful analysis and testing, I believe this problem has been adequately addressed.

Of course, are free to try the changes in your test environment now.  Simply override the SDG dependency (using Maven or Gradle) pulled in by SBDG (e.g. if you are using the starter) to include the `org.springframework.data:spring-data-geode:1.6.7-SNAPSHOT`, which will contain the changes.

If you have other questions, let me know.

If you encounter other issues, please file a new ticket in SDG, [here](https://github.com/spring-projects/spring-data-geode/issues).

**Comment by janmaslik** [comment]
Hi jxblum ,
Thank you very much for detailed explanation and support. I try to use the property and later i will try new version.

**Comment by jxblum** [comment]
You are welcome. Sounds good and if you have any more problems, please respond here or open a new ticket. Also, just a friendly reminder, the fix for this will be available in SBDG `1.6.12`, `1.7.4` and `2.0.0-M5`.

Cheers!
